lore/developers/contributing/designs/network-design.md

```diff
@@ -0,0 +1,76 @@
+
+### Floating IPs
+
+Some of the hosts running on the control plane will need to be accessed from the outside world for various reasons.  Public addresses for floating IPs will be allocated from the ``external`` network that is available to our cloud, currently providing addresses from the 169.44.171.64/27 range, though our quota currently allows for 50. Each of the following hosts will also be assigned a floating IP from the external network that will be publically accessible on specific ports (see [Security Groups](#security-groups)). Public DNS hostnames will be updated accordingly:
+
+ - bastion.bonnyci-internal.portbleu.com <- contrasjc-bastion.portbleu.com - Provides an ssh entry point from the outside world to the rest of our infrastructure and serves automation logs via http.
+ - logs.bonnyci-internal.portbleu.com <- logs.bonnyci.com - Serves test log output and other test archives via http.
+ - zuul.bonnyci-internal.portbleu.com <- zuul.bonnyci.org - Provides a https endpoint for receiving Github webhook notifications, and the serves Zuul status JSON and web site.
+
+### Security Groups
+
+Security groups will be used to limit the connectivity from the outside world to only specific ports for the services described above.  They will also be used to classify and limit internal connectivity between internal networks.  Traffic on any port not explicitly allowed will be denied.
+
+The following security groups will be created on the ``control-plane`` project:
+
+* ``sg-control-plane`` - This group will allow all traffic from the other hosts that are part of the same ``sg-control-plane`` security group.
+- Ports open: All ports from security group ``sg-control-plane``.
+
+* ``sg-zuul-merger`` - This group will allow git to be served to nodepool slaves.
+- Ports open:  8858 from network 10.0.0.0/8
+
+* ``sg-ssh`` - A generic group that allows unrestricted inbound ssh access.
+- Ports open: 22 (ssh) from 0.0.0.0/0
+
+* ``sg-http-https`` - A generic group that allows unrestricted inbound http and https access.
+- Ports open: 80 (http) from 0.0.0.0/0, 443 (https) from 0.0.0.0/0
+
+The following hosts will have the corresponding security groups assigned:
+
+* bastion.bonncyi-internal.portbleu.com: sg-ssh, sg-http-https, sg-control-plane
+* logs.bonnyci-internal.portbleu.com: sg-control-plane, sg-http-https
+* merger01.bonnyci-internal.portbleu.com: sg-control-plane, sg-zuul-merger
+* nodepool.bonncyi-internal.portbleu.com: sg-control-plane
+* zuul.bonnyci-internal.portbleu.com: sg-control-plane, sg-http-https
+
+SSH access to the entire infrastructure will need to happen via the bastion host as it is the only one capable of accepting SSH connections from the outside world.
+
+The ``nodepool`` project will boot all of its nodes on the default security group.  This group will need to be updated to allow unrestricted access from the ``192.168.10.0/24`` network.
```

Comment:
Unfortunately, it looks like you cannot specify the security groups that nodepool uses to boot its instances, so we're stuck using the default group.  pabalenger mentioned this would cause breakage across clouds in openstack-infra but I'm not certain of the specifics.   We could limit it to accepting only SSH traffic since AFAIK thats the only method nodepool/zuul will use to do anything with the slaves.